girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -303,7 +315,12 @@ def deleteFile(self, file):
         if matching.count(True) == 1:
             path = os.path.join(self.assetstore['root'], file['path'])
             if os.path.isfile(path):
-                os.remove(path)
+                with self.deleteLock:
+                    if self.model('upload').findOne(q) is None:
```

Comment:
Fair point. The number of uploads in practice will increase over time since we don't have a built-in process for cleaning up partial ones automatically, but we do have an endpoint that does so.